documentation/design_proposal/002_grafana_v1beta1.md

```diff
@@ -0,0 +1,191 @@
+}
+
+// GrafanaContainer provides a means to configure the grafana container
+type GrafanaContainer struct {
+ BaseImage         string                   `json:"baseImage,omitempty"`
+ Resources         *v1.ResourceRequirements `json:"resources,omitempty"`
+ ReadinessProbe    *v1.Probe                `json:"readinessProbe,omitempty"`
+ LivenessProbeSpec *v1.Probe                `json:"livenessProbe,omitempty"`
+}
+```
+
+Instead of setting `grafana.spec.baseImage` you would have to do `grafana.spec.grafanaContainer.baseImage`.
+
+Checkout the [grafana_types](002_grafana_types.go) file for a better example. Just overwrite `api/integreatly/v1alpha1/grafana_types.go` with this file.
+
+### Change in defaults
+
+#### Default Security Context
+
+Today we have a number of defaults that would be good to apply to the operator it self and the grafana deployment
+to make it as secure as possible out of the box. One of the problems with this in openshift [SCC](https://docs.openshift.com/container-platform/4.9/authentication/managing-security-context-constraints.html)
+and we have to add some check to see if we are deploying in a openshift environment or not.
+
+One of the configs we should asspire to is something like this in our container seucirty context.
+Openshift don't support providing `runAsGroup` and `runAsUser`, `runAsNonRoot`might work since it gets user and group from openshift by default.
+
+> These settings should also apply to our plugin initContainer.
+
+```.yaml
+    securityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop:
+        - NET_RAW
+      privileged: false
+      readOnlyRootFilesystem: true
+      runAsGroup: 1000
+      runAsNonRoot: true
+      runAsUser: 1000
+```
+
+These settings also forces us to add some default volumes else grafana won't start. TODO add the volumes needed.
+
+#### Default routes/ingress
+
+Today it's not possible to define none https route. By seperating routes and ingress in to seperate configs in the crd we should be able to seperate these defaults easier.
+In openshift a good default is TLS enabled by default. While in ingress it's not as easy.
+
+So what needs to change?
+
+The default setting for routes should be [edge](https://docs.openshift.com/container-platform/4.9/networking/routes/secured-routes.html) `TLS enabled=true` but enable the possability to run in http mode only.
+
+The default ingress should keep on being TLS enabled=false
+
+#### Default resource requests
+
+To miniamize the risk of our operator and the grafana deployment to be evicted during high load in your clusters we should provide basic resource reuqests for both the operator and the grafana deployment/init container.
+
+## Alternatives
+
+Instead of having a custom object for grafanaContainer we could provide the user with the entire deployment object with some tweaks and have them configure everything from inside the deployment.
```

Comment:
I like this idea as well, again, this would prevent us from making those assumptions for the user and just letting them set it up themselves, for fields that we deem reasonable